Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp

```diff
@@ -24,83 +24,98 @@
  */
 
 #include "config.h"
-#include "SharedRingBufferStorage.h"
 
 #if USE(MEDIATOOLBOX)
+#include "SharedCARingBuffer.h"
 
 #include <WebCore/CARingBuffer.h>
 
 namespace WebKit {
 
-ReadOnlySharedRingBufferStorage::ReadOnlySharedRingBufferStorage(const SharedMemory::Handle& handle)
-    : m_storage(SharedMemory::map(handle, SharedMemory::Protection::ReadOnly))
-{
-}
-
-void* ReadOnlySharedRingBufferStorage::data()
+void* SharedCARingBufferBase::data()
 {
     return m_storage ? static_cast<Byte*>(m_storage->data()) + sizeof(FrameBounds) : nullptr;
 }
 
-auto ReadOnlySharedRingBufferStorage::sharedFrameBounds() const -> const FrameBounds*
-{
-    return m_storage ? reinterpret_cast<const FrameBounds*>(m_storage->data()) : nullptr;
-}
-
-auto ReadOnlySharedRingBufferStorage::sharedFrameBounds() -> FrameBounds*
+auto SharedCARingBufferBase::sharedFrameBounds() const -> FrameBounds*
 {
-    return m_storage ? reinterpret_cast<FrameBounds*>(m_storage->data()) : nullptr;
+    return reinterpret_cast<FrameBounds*>(m_storage->data());
```

Comment:
Other methods (e.g. `SharedCARingBufferBase::data()`, `SharedCARingBufferBase::size()`, etc) still null check `m_storage`, is there really no need to do so here?

Reply:
Great eyes. Yes, went ahead of myself and then backtracked and then made the mistake.
Added the null check.